assembly/compliance-tool/src/main/java/org/eclipse/sw360/antenna/frontend/compliancetool/sw360/status/SW360StatusReporterParameters.java

```diff
@@ -0,0 +1,139 @@
+            throw new IllegalArgumentException(
+                    "The provided parameter " + parameter + " does not adhere to the structure --<parameter>=<value> and is therefore invalid.");
+        }
+        return parameterParts[1];
+    }
+
+    /**
+     * Parses a parameter key from a parameter key value pair separated by {@code AbstractAntennaCLIOptions.PARAMETER_IDENTIFIER}
+     * @param parameter String that contains a parameter key and a value plus identifiers.
+     * @return String representing a parameter key
+     * @throws IllegalArgumentException if more than one or no
+     * {@code AbstractAntennaCLIOptions.PARAMETER_IDENTIFIER} is used.
+     */
+    private static String getParameterKeyFromParameter(String parameter) {
+        String[] parameterParts = parameter.split(AbstractAntennaCLIOptions.PARAMETER_IDENTIFIER);
+        if (parameterParts.length != 2) {
+            throw new IllegalArgumentException(
+                    "The provided parameter " + parameter + " does not adhere to the structure --<parameter>=<value> and is therefore invalid.");
+        }
+        return parameterParts[0];
+    }
+
+    /**
+     * Parses the parameter value of a parameter key from a set of parameters.
+     * @param parameters Set of parameters searched for parameter key
+     * @param parameterKey parameter key of the parameter key value pair
+     * @return parameter value of the parameter key if contained in parameter set, otherwise null
+     */
+    static String parseParameterValueFromListOfParameters(Set<String> parameters, String parameterKey) {
+        return parameters.stream()
+                .filter(p -> getParameterKeyFromParameter(p).equals(parameterKey))
+                .findFirst()
+                .map(SW360StatusReporterParameters::getParameterValueFromParameter)
+                .orElse(null);
+    }
+
+    /**
+     * Returns a help message that describes the parameter options supported
+     * by the status reporter.
+     *
+     * @return the help message
+     */
+    static String helpMessage() {
+        String cr = System.lineSeparator();
+        return "Usage: java -jar compliancetool.jar " + AntennaComplianceToolOptions.SWITCH_REPORTER + "[options] <complianceMode> <propertiesFilePath>" + cr + cr +
+                "The reporter create a csv file for every run with the requested information statement. " + cr + cr +
+                "The reporter info statements: (only one can be set)" + cr +
+                new IRGetClearedReleases().getInfoParameter() + ":   Gives a list of all releases in a given sw360 instances that are cleared." + cr +
+                new IRGetNotClearedReleases().getInfoParameter() + ":   Gives a list of all releases in a given sw360 instances that are not cleared." + cr +
+                new IRGetReleasesOfProjects().getInfoParameter() + ":   Gives a list of all releases of a given project in a given sw360 instances." + cr;
+    }
```

Comment:
The information which concrete reports are supported is distributed over multiple classes; here the parameters class has to know all supported report types in order to generate the help message. Only the InfoRequestFactory class should know this. So either InfoRequestFactory could generate the help message or it could provide a method which returns a collection of all supported InfoRequest objects; from this collection the help text can then be generated.